software/control/gui_hcs.py

```diff
@@ -679,6 +689,10 @@ def makeConnections(self):
 
                 self.multipointController.napari_layers_init.connect(self.napariMultiChannelWidget.initLayers)
                 self.multipointController.napari_layers_update.connect(self.napariMultiChannelWidget.updateLayers)
+                self.napari_connections['napariMultiChannelWidget'] = [
```

Comment:
Instead of keeping track of the napari_connections, It might be a bit easier, and more generalizable to other performance fixes to instead just make all the performance-mode-only connections and disconnections in a single method.  As one possible way of doing this, we could get rid of `self.nepari_connections` and have something like your `setup*Connections` method, and move all the `connect`/`disconnect` there.  For instance:

```
def updatePerformanceWidgetConnections():
    if self.performance_mode:
        # Disconnect all the performance mode widgets from the multipointController
        self.multipointController.napari_layers_init.disconnect(self.napariMultiChannelWidget.initLayers)
        self.multipointController.napari_layers_update.disconnect(self.napariMultiChannelWidget.updateLayers)
        # ... the same for all the other performance mode widgets
    else:
       # Not in performance mode, setup all the connections to the above widgets
        self.multipointController.napari_layers_init.connect(self.napariMultiChannelWidget.initLayers)
        self.multipointController.napari_layers_update.connect(self.napariMultiChannelWidget.updateLayers)
       # ... all the others ...
```

Then we just call self.updatePerformanceModeConnections() in `__init__` and when we get a performance mode toggle update. 

There might be much nicer ways to do this in qt as well.  For instance, if you're in c++ it looks like `QMetaObject::disconnect` might be nice - it disconnects all connections between two `QObject`.  There also might be a way to block signals from one QObject to another.  I'm not too qt familiar, though, so would need to investigate more!